experimental/HeapSparseLA/src/Types.jl

```diff
@@ -0,0 +1,63 @@
+
+mutable struct HeapNode{T}
+  content::T
+  index::Int
+  cofactor::Union{T, Nothing} # an element !==nothing here indicates
+                              # that the contents here and all of the 
+                              # following subtree shall be multiplied 
+                              # with `cofactor` to get the actual values.
+  left::Union{HeapNode{T}, Nothing}
+  right::Union{HeapNode{T}, Nothing}
```

Comment:
Thank you for the idea. Can't we store a row index-cofactor pair in a vector, or can't pairs be stored contiguously? I'm not very experienced with memory layout in Julia.

Reply:
I am happy to play, but few questions:
  - what operations are supported?
  - should this not be done in Hecke, where the sparse stuff is, or in AA where it should be?
  - how does it compare with magma? The in my experience fastest sparse Lin alg available in general - and not using plain Gauss
  - is this any good at finding kernel vectors?
I am in favour of speeding up Lin alg, as this the bottleneck for cohomology....
Not over F_2 though

Reply:
> what operations are supported?
Probably the usual and `pivot`, `pop!`, and friends on the rows? I think, it might also be nice to have support for lazy evaluation of linear combinations. 

> should this not be done in Hecke, where the sparse stuff is, or in AA where it should be?

If we get somewhere, then probably yes. We can move it later, right?

> is this any good at finding kernel vectors?

No idea. Actually I don't even recall how kernels are computed over the integers.